plugins/aws/aws.plugin.zsh

```diff
@@ -39,73 +40,79 @@ function acp() {
     return 1
   fi
 
-  local aws_access_key_id="$(aws configure get aws_access_key_id --profile $1)"
-  local aws_secret_access_key="$(aws configure get aws_secret_access_key --profile $1)"
-  local aws_session_token="$(aws configure get aws_session_token --profile $1)"
-  local mfa_serial="$(aws configure get mfa_serial --profile $1)"
-  local role_arn="$(aws configure get role_arn --profile $1)"
+  local profile="$1"
+
+  # Get fallback credentials for if the aws command fails or no command is run
+  local aws_access_key_id="$(aws configure get aws_access_key_id --profile $profile)"
+  local aws_secret_access_key="$(aws configure get aws_secret_access_key --profile $profile)"
+  local aws_session_token="$(aws configure get aws_session_token --profile $profile)"
+
 
   # First, if the profile has MFA configured, lets get the token and session duration
-  local mfa_opt=""
+  local mfa_serial="$(aws configure get mfa_serial --profile $profile)"
 
-  if [[ -n $mfa_serial ]]; then
-    local mfa_token=""
-    echo "Please enter your MFA token for $mfa_serial:"
+  if [[ -n "$mfa_serial" ]]; then
+    local -a mfa_opt
+    local mfa_token sess_duration
+    echo -n "Please enter your MFA token for $mfa_serial: "
     read -r mfa_token
```

Comment:
```suggestion
    vared -p "Please enter your MFA token for $mfa_serial: " -c mfa_token
```
I'd suggested using built-in `vared` instead of the combination of `echo` and `read`.
As fo me, it looks pretier:

before:
```
Please enter your MFA token for arn:aws:iam::11111111111:mfa/user:
12346
```
after:
```
Please enter your MFA token for arn:aws:iam::11111111111:mfa/user: 12346
```

Reply:
Note that this uses `echo -n` which doesn't add the newline, so it's effectively the same.